File: gant-proto/src/composable/featureOption.ts
import {getFeatureOptions} from "@/composable/auth";
import {FeatureOption} from "@/const/common";


export function available(optionName: FeatureOption) {

    const featureOptions = getFeatureOptions()

    return featureOptions?.find(v => v.name == optionName && v.enabled) != undefined
}

File: backend/api/utils/feature_option.go
package utils

import (
	"github.com/kenkonno/gantt-chart-proto/backend/api/constants"
	"github.com/kenkonno/gantt-chart-proto/backend/models/db"
	"github.com/kenkonno/gantt-chart-proto/backend/repository"
	"github.com/samber/lo"
)

func HasOption(optionName constants.Feature) bool {

	featureOptionRep := repository.NewFeatureOptionRepository()

	_, exists := lo.Find(featureOptionRep.FindAll(), func(item db.FeatureOption) bool {
		return optionName.Equals(item.Name) && item.Enabled == true
	})

	return exists
}

func GetProjectSortKey() string {
	if HasOption(constants.ProjectListNameSort) {
		return "name"
	} else {
		return "id"
	}
}
